kylin-spark-project/kylin-spark-engine/src/main/scala/org/apache/kylin/engine/spark/job/CubeBuildJob.java

```diff
@@ -157,7 +157,12 @@ protected void doExecute() throws Exception {
                 logger.info("Triggered cube planner phase one .");
         }
 
-        buildLayoutWithUpdate = new BuildLayoutWithUpdate();
+        if (config.isReuseParentDataSource()) {
+            buildLayerWithUpdate = new BuildLayerWithUpdate();
+        } else {
+            buildLayoutWithUpdate = new BuildLayoutWithUpdate();
```

Comment:
Why needs these two similar Class? It's better to refactor the original class 'BuildLayoutWithUpdate' to support this feature.